samcli/lib/utils/resource_trigger.py

```diff
@@ -182,6 +183,10 @@ def __init__(
         self._on_code_change = on_code_change
         self.base_dir = base_dir
 
+        self._watch_exclude = [*DEFAULT_WATCH_IGNORED_RESOURCES]
+        for exclude in watch_exclude or []:
+            self._watch_exclude.append(f"^.*{exclude}.*$")
```

Comment:
Do we need to wrap it with regex, can't we just let customers to define their own input?

Reply:
Did you mean letting customers define their own regex? We can do that, but it'll mean that the customer has to make sure they match the parent path correctly so that the resource is ignored

Reply:
Let's get this out like this, if customers want to add regex from their side, we can add this prefix & suffix conditionally.